### PR TITLE
fix: fixed data aggregation plugin

### PR DESCRIPTION
**Pull request by zefirka**

### CODE REVIEW

File: src/plugins/dataRefs/dataRefs.ts [hunk 1]
```diff
@@ -176,16 +84,16 @@ const DataRef = (opst: DataRefsPluginOptions) => {
             calcRefs: (fromIdx: number, toIdx: number, seriesId: string) => {
                 const seriesIdx = _.state.y2uIdx[seriesId];
                 const timestamps = _.uplot.data[0].slice(fromIdx, toIdx + 1) as number[];
-                const values = _.uplot.series[seriesIdx].$c.slice(fromIdx, toIdx + 1) as (number | null)[];
+                const values = _.uplot.series[seriesIdx].$c;
```
Comment (astandrik) [inline src/plugins/dataRefs/dataRefs.ts:87]: Would you kindly give $c a human readable name?

As well as I see it's  $c: rawSeries.data,

File: tests/plugins/dataRefs.test.ts [hunk 2]
```diff
@@ -186,4 +186,55 @@ describe('DataRefs plugin', () => {
...
+
+    describe('data with strings', () => {
+        const y = new Yagr(window.document.createElement('div'), {
+            timeline: [1, 2, 3, 4],
+            scales: {
+                y: {stacking: true},
+            },
+            series: [
+                {data: [1, 'x', 3, 4], scale: 'y', id: 'one'},
+                {data: [2, 'x', null, 3], scale: 'y', id: 'two'},
```
Comment (astandrik) [inline tests/plugins/dataRefs.test.ts:198]: Could you explain whats the difference between 'x' and null values?

File: src/plugins/dataRefs/utils.ts [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        x1 = Number(values[i]);
+        t1 = timestamps[i];
+
+        // we skip over all non-numeric values in a serie
+        // so that all holes and single points (surrounded by
+        // empty void) do not add anything to the integral
+        if (!Number.isNaN(x1) && !Number.isNaN(x0)) {
+            const dt = t1 - t0;
+            const dx = x1 - x0;
+            const area = ((x0 + dx / 2) * dt) / 1000; // convert milliseconds to seconds
```
Comment (astandrik) [inline src/plugins/dataRefs/utils.ts:22]: So the integral for

values = [1, 0, -1]
timestamps = [1, 2, 3]

will be 0


Is this expected value?

probably should use absolute values?

File: src/plugins/dataRefs/utils.ts [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        }
+
+        t0 = t1;
+        x0 = x1;
+    }
+
+    return integral;
+}
+
+export function count(values: DataSeriesExtended, from = 0, to = values.length - 1) {
```
Comment (astandrik) [inline src/plugins/dataRefs/utils.ts:33]: countNumbers?

**Review by astandrik — APPROVED** [review]

File: src/plugins/dataRefs/dataRefs.ts (hunk 1 above)
Comment (zefirka) [reply src/plugins/dataRefs/dataRefs.ts:87]: I will be a breaking change. Planning for v4

File: tests/plugins/dataRefs.test.ts (hunk 2 above)
Comment (zefirka) [reply tests/plugins/dataRefs.test.ts:198]: https://yagr.tech/en/api/data-processing#problem

**Pull request merged**